build-tools/packages/build-cli/src/handlers/checkFunctions.ts

```diff
@@ -931,21 +932,12 @@ export const checkCompatLayerGeneration: StateHandlerFunction = async (
 		return true;
 	}
 
-	// layerGeneration:gen should be run from the root. It will only update packages that have the layerGeneration:gen
-	// script defined in their package.json.
-	const result = await execa.command(`pnpm run -r layerGeneration:gen`, {
-		cwd: context.root,
-	});
-	log.verbose(result.stdout);
+	const isUpToDate = await runCompatLayerGenerationCheck(context.fullPackageMap.values());
```

Comment:
This should only run on packages in the release group, or individual packages depending on the argument. We should never pass the full list of packages in the repo.